core/net/options.go

```diff
@@ -25,24 +26,54 @@ func WithThreadKey(key thread.Key) KeyOption {
 // If this is just a public key, the service itself won't be able to create records.
 // In other words, all records must be pre-created and added with AddRecord.
 // If no log key is provided, one will be created internally.
-func WithLogKey(key crypto.Key) KeyOption {
-	return func(args *KeyOptions) {
+func WithLogKey(key crypto.Key) NewThreadOption {
+	return func(args *NewThreadOptions) {
 		args.LogKey = key
 	}
 }
 
+// WithNewThreadToken provides authorization for creating a new thread.
+func WithNewThreadToken(t thread.Token) NewThreadOption {
```

Comment:
All these options get much more simplified now. The API methods only ever need to take the token string.